File: lw7/MotorinaEvgeniya/lw7/lw7/ThreadPool.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lw7
{
    class ThreadPool
    {
        private List<Task> m_threads = new List<Task>();

        public void AddThread(Task task)
        {
            m_threads.Add(task);
        }

        public void WaitThread()
        {
            foreach(Task task in m_threads)
            {
                task.Wait();
            }
        }
    }
}

File: lw6/Motorina_Eugenia/lab6/lab6/ThreadPool.cpp
#include "stdafx.h"
#include <thread>
#include "ThreadPool.h"


ThreadPool::ThreadPool()
{
}

ThreadPool::~ThreadPool()
{
}

void ThreadPool::AddThread(std::thread &&thread)
{
	m_threads.push_back(std::move(thread));
}

void ThreadPool::WaitThread()
{
	for (auto &thread : m_threads)
	{
		if (thread.joinable())
		{
			thread.join();
		}
	}
}